src/System.Private.DataContractSerialization/src/System/Runtime/Serialization/DataContract.cs

```diff
@@ -657,55 +658,67 @@ internal static int GetId(RuntimeTypeHandle typeHandle)
             // check whether a corresponding update is required in ClassDataContract.IsNonAttributedTypeValidForSerialization
             private static DataContract CreateDataContract(int id, RuntimeTypeHandle typeHandle, Type type)
             {
-                lock (s_createDataContractLock)
+                DataContract dataContract = s_dataContractCache[id];
```

Comment:
My bad I looked at the wrong one. The s_dataContractCache is an array and not a ```Dictionary<...>```. The id parameter was computed in DataContractCriticalHelper.GetId which utilizes a lock so I think this is thread-safe.